nats.go

```diff
@@ -2518,13 +2544,71 @@ func (nc *Conn) Publish(subj string, data []byte) error {
 	return nc.publish(subj, _EMPTY_, data)
 }
 
+// Used to create a new message for publishing that will use headers.
+func NewMsg(subject string) *Msg {
+	return &Msg{
+		Subject: subject,
+		Header:  make(http.Header),
+	}
+}
+
+const hdrMagic = "⚡NATS⚡/0.1\r\n"
```

Comment:
this seems pretty good in general except for the little lightnings :) telnet protocol demos nicely, this makes it hard

Reply:
We need something unique to determine that this message has headers and we are text based so needed something like this. Have seen others doing it as well.

You going to put headers into a message via telnet?

Reply:
Probably not, but would we want to make it impossible?

Reply:
Not sure if telnet can send utf8

Reply:
It's a minor nit, just remember reading the protocol spec and it made a point of being clear text and easy to play with over something like telnet - in reality thats useless, but its a good philosophy about the protocol as it has implications for how light weight clients can be etc

Reply:
Yea I am playing with it, does not work yet but I think if I use escape codes instead of utf8 directly it may. Will report back.

Reply:
Sounds like clients will need to indicate they can handle headers via CONNECT anyway, meaning telnet sessions etc will most likely not send or receive headers.

Reply:
We moved this to new proto. Will update this branch to new protos.